src/Lookup/InProcessCachingDataTypeLookup.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\PropertyId;
+
+/**
+ * PropertyDataTypeLookup that provides in-process caching.
+ *
+ * @since 3.1
+ *
+ * @licence GNU GPL v2+
+ * @author Katie Filbert < aude.wiki@gmail.com >
+ */
+class InProcessCachingDataTypeLookup implements PropertyDataTypeLookup {
```

Comment:
Sorry, InMemoryDataTypeLookup is not about caching. What about a CachingPropertyDataTypeLookup that gets a HashBagOStuff via constructor? Isn't this exactly what the BagOStuff is about?